include/nbl/builtin/hlsl/blur/box_blur.hlsl

```diff
@@ -0,0 +1,318 @@
+#pragma once
+
+#include <nbl/builtin/hlsl/blur/common.hlsl>
+
+#include <nbl/builtin/hlsl/glsl_compat/core.hlsl>
+#include <nbl/builtin/hlsl/workgroup/arithmetic.hlsl>
+#include <nbl/builtin/hlsl/workgroup/scratch_size.hlsl>
+
+
+// https://github.com/microsoft/DirectXShaderCompiler/issues/6144
+static nbl::hlsl::uint32_t3 /*nbl::hlsl::glsl::*/gl_WorkGroupSize() { return uint32_t3( WORKGROUP_SIZE, 1, 1 ); }
+
+static const uint32_t ITEMS_PER_THREAD = ( AXIS_DIM + WORKGROUP_SIZE - 1 ) / WORKGROUP_SIZE;
+static const uint32_t ITEMS_PER_WG = ITEMS_PER_THREAD * WORKGROUP_SIZE;
```

Comment:
you're computing it wrong, the arithmetic will always prefix sum `WORKGROUP_SIZE` elements, because you'll join up the prefix sums yourself

Now you're making us use more shared memory than is necessary.

Reply:
Also I'm not sure you need `ITEMS_PER_THREAD` anywhere.